src/Psalm/Internal/Algebra/FormulaGenerator.php

```diff
@@ -153,7 +153,6 @@ public static function getFormula(
                         $redefined = false;
 
                         if ($var[0] === '=') {
-                            /** @var string */
```

Comment:
This is only possible because you set composer to PHP 8.1. We decided that Psalm must still be able to run on PHP 7.4 for its version 5.

You should revert the change on composer to still allow 7.4, otherwise this won't be merged in Psalm 5 (and will probably won't be released soon: allowing only PHP8.1 would mean denying two third of our users)

Reply:
I'll revert this, but still, I see nothing wrong with restricting psalm to 8.1, since it's a development tool that is only ever required as a development dependency and runs mainly in CI jobs, where the PHP version isn't as important and doesn't require changes to production code.

Reply:
I don't have stats for that, but there is a lot of Psalm runs that are made manually by developpers locally. Once you take that into account, you have to factor in every developer config. For example, for a dev on windows running Wamp to run their application on their laptop, installing a new PHP version just for Psalm may be a real pain.

Psalm may also need to use reflection on particular extensions that may only be available for certain PHP versions. 

We made a big leap to 7.4 for Psalm 5 (remember Psalm 4 supported PHP 7.1. It was initially 7.3 but users complained). It still represents about 6% of Psalm installs to this day that we're gonna left behind. We feel it's acceptable, but we're not ready to dump 66% (not even if we could say that they can easily change their CI jobs)

Reply:
That's understandable (especially the windows part), I'll revert this here and keep using 8.1 in my own fork, I really want to use readonly properties to catch eventual runtime corruptions :)